crime-commons-classes/src/main/java/uk/gov/justice/laa/crime/enums/CrownCourtOutcomeType.java

```diff
@@ -0,0 +1,24 @@
+package uk.gov.justice.laa.crime.enums;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.stream.Stream;
+
+@Getter
+@AllArgsConstructor
+public enum CrownCourtOutcomeType {
+    TRIAL("TRIAL"), APPEAL("APPEAL");
+
+    private final String type;
+
+    public static CrownCourtOutcomeType getFrom(String type) {
+        if (StringUtils.isBlank(type)) return null;
+
+        return Stream.of(CrownCourtOutcomeType.values())
+                .filter(ccoType -> ccoType.type.equals(type))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(String.format("CrownCourtOutcomeType with value: %s does not exist.", type)));
```

Comment:
Personally I have a preference for enclosing 'untrusted' or otherwise invalid input in messages as I feel it is easier to spot some types of issues. It's a pretty edge case, so totally optional, but something to consider!

i.e. consider the following
```
    @Test
    void givenInvalidResultString_whenGetFromIsInvoked_thenExceptionIsThrown() {
       CrownCourtOutcome.getFrom("MOCK_RESULT_STRING exists\n\n\n\n\n\n\n");
    }
```
Will throw an exception with the message:
```
java.lang.IllegalArgumentException: Crown Court Outcome with value: MOCK_RESULT_STRING exists






 does not exist.
```
Which if not properly escaped (and/oror enclosed) could be misleading at first glance, compared to
```
java.lang.IllegalArgumentException: Crown Court Outcome with value: [MOCK_RESULT_STRING exists






] does not exist.
```

Reply:
Ok a useful thing to remember thanks! Enclosed the untrusted input in the exception message above 👍